docs/PLUGIN_AUTHORING.md

```diff
@@ -17,152 +17,134 @@ See the License for the specific language governing permissions and
-Snap supports three type of plugins. They are collectors, processors, and publishers.  The plugin project name should use the following format:  
->snap-plugin-[type]-[name]
+A few notes before we get started:
+* Before writing a new Snap plugin, please check out the [Plugin Catalog](./PLUGIN_CATALOG.md) to see if any existing plugins meet your needs.
+* If you need any assistance, please reach out on [slack #snap-developers channel](https://intelsdi-x.herokuapp.com/).
 
-For example:  
->snap-plugin-collector-hana      
->snap-plugin-processor-movingaverage    
->snap-plugin-publisher-influxdb  
+## Developing Plugins
 
-Example files and directory structure:  
-```
-snap-plugin-[type]-[name]
- |--[name]
-  |--[name].go  
-  |--[name]_test.go  
-  |--[name]_integration_test.go
- |--main.go
- |--main_test.go
-```
+### Plugin Type
 
-### Metric Naming
-A plugin should **NOT** advertise metrics which namespaces contain:
+Snap supports three type of plugins:
 
-##### a) the following characters in a namespace:
-    - spaces	` `
-    - brackets: `()[]{}`
-    - slashes:  `| \ /`
-    - carets:   `^`
-    - quotations:   `" ' \``
-    - other punctuations: `. , ; ? !`
+* collector: gathering metrics
+* processor: transforming metrics
+* publisher: publishing metrics
 
-##### b) a wildcard in the end
+### Plugin Name
 
-Example:
+The plugin name should follow this format: `snap-plugin-[type]-[name]`
 
-| Unacceptable metric namespace | Why                    | Proposal                                  |
-|:------------------------------|:-----------------------|:------------------------------------------|
-| /intel/foo/\*                 | a wildcard in the end  | /intel/foo/\*/bar <br/> /intel/foo/\*/baz |
-| /intel/mock/bar(no)           | not allowed characters | /intel/mock/bar_no                        |
-| /intel/mock/bar("no")         | not allowed characters | /intel/mock/bar_no                        |
-| /intel/mock/bar^no            | not allowed characters | /intel/mock/bar_no                        |
-| /intel/mock/bar.no            | not allowed characters | /intel/mock/bar_no                        |
-| /intel/mock/bar!?             | not allowed characters | /intel/mock/bar                           |
+For example:
+* `snap-plugin-collector-hana`
+* `snap-plugin-processor-movingaverage`
+* `snap-plugin-publisher-influxdb`
 
+### Plugin Metric Namespace
 
-Snap validates the metrics exposed by the plugin and, if validation fails, an error is returned and the plugin is not loaded.
+When gathering data in collector plugins, each metric requires a namespace and a description of what data is being gathered. The metric namespace should contain the org, name of plugin, and the metric's name:
 
-##### c) static and dynamic metrics
-Snap supports both static and dynamic metrics.  You can find more detail about static and dynamic metrics [here](./METRICS.md).
+`/[organization]/[plugin name]/[plugin namespace]/[metric name]`
```

Comment:
I think we should specify that <metric name> can be multiple sections. This kind've reads like it's always 4.